idunn/blocks/images.py

```diff
@@ -0,0 +1,131 @@
+import re
+import logging
+import hashlib
+import urllib.parse
+
+from apistar import types, validators
+from .base import BaseBlock, BlocksValidator
+from .wikipedia import WikidataConnector, WikiUndefinedException
+
+logger = logging.getLogger(__name__)
+
+class ThumbrHelper:
+    _salt = None
+    _sub_domains = None
+    _use_thumbr = None
+
+    @classmethod
+    def get_salt(cls):
+        if cls._salt is None:
+            from app import settings
+            cls._salt = settings.get('THUMBR_SALT')
+        return cls._salt
+
+    @classmethod
+    def use_thumbr(cls):
+        if cls._use_thumbr is None:
+            from app import settings
+            cls._use_thumbr = settings.get('USE_THUMBR')
+        return cls._use_thumbr
+
+    @classmethod
+    def init_sub_domains(cls):
+        if cls._sub_domains is None:
+            from app import settings
+            cls._sub_domains = settings.get('THUMBR_DOMAINS').split(',')
+
+    @classmethod
+    def get_domain(cls, hash):
+        cls.init_sub_domains()
+        n = int(hashlib.sha256(bytes(hash[0], encoding='utf8')).hexdigest(), 16) % (len(cls._sub_domains))
+        return 'https://' + cls._sub_domains[n] + '/thumbr'
+
+    @classmethod
+    def get_url_remote_thumbnail(cls, source, width=0, height=0, bestFit=True, progressive=False, animated=False):
+        displayErrorImage = False
+
+        salt = cls.get_salt()
+        token = f"{source}{width}x{height}{salt}"
+        hash = hashlib.sha256(bytes(token, encoding='utf8')).hexdigest()
+        domain = cls.get_domain(hash)
+
+        size = int(width) * int(height)
+
+        hashURLpart = f"{hash[0]}/{hash[1]}/{hash[2:]}"
+
+        filename = source.split('/')[-1]
+        if not bool(re.match("^.*\.(jpg|jpeg|png|gif)$", filename)):
+            filename += ".jpg"
+
+        params = f"u={urllib.parse.quote(source)}&q={1 if displayErrorImage else 0}&b={1 if bestFit else 0}&p={1 if progressive else 0}&a={1 if animated else 0}"
```

Comment:
Again, that needs to be tested. But according to https://docs.python.org/3/library/urllib.parse.html#url-quoting `quote` does not encode `/` characters. Is that a problem ? 
Using [`urllib.parse.urlencode`](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlencode) (that accepts a dict as input) may be more suitable